modules/statistic.rb

```diff
@@ -0,0 +1,26 @@
+module Statistic
+  def often_take_book_reader(orders, popular_readers = 1)
+    group_sort(orders, :reader, popular_readers)
+  end
+
+  def most_popular_book(orders, popular_book_count = 3)
+    group_sort(orders, :book, popular_book_count)
+  end
+
+  def readers_three_most_popular_books(orders, popular_book_count = 3)
+    popular_orders = group_sort(orders, :book).take(popular_book_count).map { |book_orders| book_orders[1] }
+    popular_orders.flatten.map(&:reader).uniq.size
```

Comment:
Maybe better to receive `reader` when you are mapping `book orders`. So you can decrease count of iterations

```ruby
popular_readers = group_sort(orders, :book).take(popular_book_count).map { |book_orders| book_orders[1].reader }
popular_readers.flatten.uniq.size
```